### PR TITLE
PMM-7422 add toggle to show silenced alerts in the failed checks table

### PR DESCRIPTION
**Pull request by nicolalamacchia**
https://github.com/Percona-Lab/pmm-submodules/pull/1452

### CODE REVIEW

File: pmm-app/src/pmm-check/Check.service.ts [hunk 1]
```diff
@@ -75,6 +82,7 @@ export const processData = (data: Alert[]): ActiveCheck[] => {
         description,
         severity: labels.severity,
         labels,
+        silenced: state === 'suppressed',
```
Comment (fabio-silva) [inline pmm-app/src/pmm-check/Check.service.ts:85]: do we have any enum for this?

File: pmm-app/src/pmm-check/Check.service.ts [hunk 2]
```diff
@@ -56,13 +56,20 @@ export const CheckService = {
...
   const result: Record<
     string,
-    Array<{ summary: string; description: string; severity: string; labels: { [key: string]: string } }>
+    Array<{
+      summary: string;
+      description: string;
+      severity: string;
+      labels: { [key: string]: string },
+      silenced: boolean;
+     }>
```
Comment (fabio-silva) [inline pmm-app/src/pmm-check/Check.service.ts:65]: Is it worth it typing this?

File: pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  test('saveShowSilencedValue calls localStorage.setItem', () => {
+    saveShowSilencedValue(true);
+
+    expect(setItemSpy).toBeCalledTimes(1);
+    expect(setItemSpy).toBeCalledWith(SHOW_SILENCED_VALUE_KEY, 'true');
+  });
+
+  test('loadShowSilencedValue calls localStorage.getItem', () => {
+    jest.spyOn(Storage.prototype, 'getItem').mockImplementation(() => 'true');
```
Comment (fabio-silva) [inline pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.test.ts:25]: Instead of repeating the whole expression on every override, maybe it's easier to:
```
let getItemSpy = jest.spyOn(Storage.prototype, 'getItem');
let setItemSpy = jest.spyOn(Storage.prototype, 'setItem');
// ...
beforeEach(() => {
  getItemSpy.mockImplementation(() => 'true');
});
// ...
test('test', () => {
 // getItemSpy.mockImplementation(...)
});
```

Comment (fabio-silva) [reply pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.test.ts:25]: And this mock in particular is already in `beforeEach`

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: pmm-app/src/pmm-check/Check.service.ts (hunk 1 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-check/Check.service.ts]: Good point, updated that

File: pmm-app/src/pmm-check/Check.service.ts (hunk 2 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-check/Check.service.ts:66]: That was my though as well, but it was there and I don't want to refactor that too much now : ) I just added what was missing, but I think we should consider refactoring that code in the future

File: pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.test.ts (hunk 3 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.test.ts]: Test updated

File: pmm-app/src/pmm-check/CheckPanel.tsx [hunk 4]
```diff
@@ -46,12 +46,12 @@ export const CheckPanel: FC = () => {
...
       label: Messages.failedChecksTitle,
       key: TabKeys.failedChecks,
-      component: <FailedChecksTab hasNoAccess={hasNoAccess} />,
+      component: <FailedChecksTab key="failed-checks" hasNoAccess={hasNoAccess} />,
     },
     {
       label: Messages.allChecksTitle,
       key: TabKeys.allChecks,
-      component: <AllChecksTab />,
+      component: <AllChecksTab key="all-checks" />,
```
Comment (fabio-silva) [inline pmm-app/src/pmm-check/CheckPanel.tsx:54]: Why do we need `key` here? Was it `data-qa` instead?

File: pmm-app/src/pmm-check/__mocks__/Check.service.ts [hunk 5]
```diff
@@ -56,6 +63,7 @@ export const processData = (data: Alert[]): ActiveCheck[] => {
         description,
         severity: labels.severity,
         labels,
+        silenced: state === 'suppressed',
```
Comment (fabio-silva) [inline pmm-app/src/pmm-check/__mocks__/Check.service.ts:66]: `AlertState.suppressed` ?

File: pmm-app/src/pmm-check/components/Table/TableBody.test.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ describe('TableBody::', () => {
   it('should render a table body with 3 rows', () => {
     const root = shallow(<TableBody data={activeCheckStub} />);
 
-    expect(root.find('tbody > tr').length).toEqual(4);
+    expect(root.find('tbody > tr').length).toEqual(5);
```
Comment (fabio-silva) [inline pmm-app/src/pmm-check/components/Table/TableBody.test.tsx:12]: `should render a table body with 3 rows` -> why is the test asserting 5 `tr` ?

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: pmm-app/src/pmm-check/CheckPanel.tsx (hunk 4 above)
Comment (fabio-silva) [reply pmm-app/src/pmm-check/CheckPanel.tsx:54]: (same for `FailedChecksTab `)

Comment (nicolalamacchia) [reply pmm-app/src/pmm-check/CheckPanel.tsx:54]: because those are rendered conditionally in a `map`, React complains if there's no `key`

File: pmm-app/src/pmm-check/components/Table/TableBody.test.tsx (hunk 6 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-check/components/Table/TableBody.test.tsx:12]: Removed the too specific text

File: pmm-app/src/pmm-check/__mocks__/Check.service.ts (hunk 5 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-check/__mocks__/Check.service.ts]: Done

File: pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.test.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    expect(setItemSpy).toBeCalledWith(SHOW_SILENCED_VALUE_KEY, 'true');
+  });
+
+  test('loadShowSilencedValue calls localStorage.getItem', () => {
+    expect(getItemSpy).toBeCalledTimes(1);
+    expect(loadShowSilencedValue()).toEqual(true);
+  });
+
+  test('loadShowSilencedValue the default value if localStorage is not available', () => {
+    jest.spyOn(Storage.prototype, 'getItem').mockImplementation(() => { throw Error('test error'); });
```
Comment (fabio-silva) [inline pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.test.ts:30]: Since we are testing error throwing, should we "silence" the console during tests?
I've done that [here](https://github.com/percona-platform/grafana/blob/0852d323bc2fb6608065859cbc858e5ab2541f18/public/app/features/integrated-alerting/components/AlertRuleTemplate/AlertRuleTemplate.test.tsx#L16) because it was too hard to follow the test output.

File: pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import { logger } from '@percona/platform-core';
+import { SHOW_SILENCED_VALUE_KEY, SHOW_SILENCED_DEFAULT } from './FailedChecksTab.constants';
+
+export const loadShowSilencedValue = (): boolean => {
```
Comment (tiagomotasantos) [inline pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.ts:4]: Is it really a requirement to store this value?

File: pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  try {
+    const showSilencedValue = window.localStorage.getItem(SHOW_SILENCED_VALUE_KEY);
+
+    if (showSilencedValue === null) {
+      return SHOW_SILENCED_DEFAULT;
+    }
+
+    return showSilencedValue === 'true';
```
Comment (tiagomotasantos) [inline pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.ts:12]: I would do here: `return !!showSilencedValue` and remove the condition. If `showSilencedValue` is null, undefined or false this will return false (which is the default), if it's 'true' it will return true. That would simplify this code. Also consider moving the return out of the catch.

**Review by tiagomotasantos — COMMENTED** [review]
Good job, just some considerations from my side and also the build has failing tests.

File: pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.ts (hunk 8 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.ts:4]: yes, it was discussed with @atymchuk

File: pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.ts (hunk 9 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.ts:12]: But if you store 'false', then it'd be true :)

Comment (tiagomotasantos) [reply pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.ts:12]: Indeed, thanks for pointing it, it's better this way then!

**Review by tiagomotasantos — COMMENTED** [review]
Good job! Just take care of the failing test and it's good to go from me.

**Review by tiagomotasantos — APPROVED** [review]

File: pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.test.ts (hunk 7 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-check/components/FailedChecksTab/FailedChecksTab.utils.test.ts:39]: Done!

**Review by fabio-silva — APPROVED** [review]
Nice one 👍

**Comment by nicolalamacchia** [comment]
Contained in this one: https://github.com/percona/grafana-dashboards/pull/1122
